### PR TITLE
Refactoring: find_latest_successful_artifact takes finder predicate

### PR DESCRIPTION
**Pull request by aereal**
Almost all implementations of `find_latest_successful_artifact` and `find_last_successful_artifact_with_path` are same.

### CODE REVIEW

**Comment by hagihala** [comment]
I think It would be better if the method could take a build number as an argument (and rename to `find_artifact`, though it conflicts with existing method).
Using the build number obtained [here](https://github.com/aereal/capistrano-strategy-jenkins_artifact/blob/39442f816fbaa8148e45c2a5fa0734911f5a6ddb/lib/capistrano/recipes/deploy/strategy/jenkins_artifact.rb#L78), you can avoid consistency problem.
https://github.com/aereal/capistrano-strategy-jenkins_artifact/pull/12#discussion_r93867255

**Comment by aereal** [comment]
I see, Hash from `get_last_successful_build` contains `timestamp` and `artifacts`, so all we have to do is get a build once and select an artifact such as:

```ruby
def select_artifact(artifacts)
  # ...
end

build = get_last_successful_build(...)
timestamp = build['timestamp']
artifact_url = select_artifact(build['artifacts']
```

**Comment by aereal** [comment]
@astj @hagihala @kga 

Please review this :raising_hand_man:

File: test/jenkins_api_client_job_ext_test.rb [hunk 1]
```diff
@@ -6,27 +6,20 @@
...
+      'artifacts' => [
+        { 'displayPath' => 'artifact.tar.gz', 'fileName' => 'artifact.tar.gz', 'relativePath' => 'artifact.tar.gz' },
+        { 'displayPath' => 'another.tar.gz',  'fileName' => 'another.tar.gz',  'relativePath' => 'another.tar.gz' },
+      ],
+      'url' => 'http://example.com/',
     }
 
-    got_artifact_path = api_client.job.find_last_successful_artifact_with_path(job_name, 'artifact.tar.gz')
+    got_artifact_path = JenkinsApi::Client::Job.get_artifact_url_by_build(build)
     assert_match %r{\bartifact\.tar\.gz\z}, got_artifact_path
```
Comment (hitode909) [inline test/jenkins_api_client_job_ext_test.rb:19]: I think you can assert result with `assert_equal`.

**Review by hitode909 — APPROVED** [review]
@aereal I wrote a nit comment. LGTM :shipit:

**Comment by astj** [comment]
@aereal Sorry for late response.. What is this p-r's status? still waiting another review?

**Comment by astj** [comment]
@aereal ping
(I need to touch around `deploy_at` for another reason, and it will conflict with this p-r)

**Comment by aereal** [comment]
@astj Merged! (sorry for late)

**Pull request merged**